### PR TITLE
feat: add Grok Build plugin and marketplace support

### PR DESCRIPTION
**Pull request by Misty-Star**
## What Changed

Add a native Grok Build plugin/marketplace surface so ECC can be installed with `grok plugin marketplace add` / `grok plugin install`, without changing the Claude marketplace layout.

- New `.grok-plugin/plugin.json` and `.grok-plugin/marketplace.json`. Grok rejects Claude's `"source": "./"` as an empty path, so the Grok catalog points at the whole-repo Git URL instead of a thin `plugins/ecc/` subdirectory.
- `resolveEccRoot()` and hook bootstrap honor `GROK_PLUGIN_ROOT` (then Claude/ECC aliases) and scan `~/.grok/plugins/` plus `~/.grok/installed-plugins/`.
- Hook inline locators, hook-flags, and related hook runners pick up the same env.
- README / zh-CN / hooks docs cover `--trust`, `grok plugin enable`, `ECC_HOOK_PROFILE`, and that trusted Grok installs attach root `.mcp.json` (`chrome-devtools`).
- Catalog sync, adapter compliance matrix, and plugin-manifest tests cover the new files.

## Why This Change

Grok Build already discovers Claude plugin directories, but its marketplace scanner drops `source: "./"`, and ECC's hook root resolver only searched `~/.claude`. Without a Grok-specific catalog and `GROK_PLUGIN_ROOT` resolution, ECC cannot be installed or run as a trusted Grok plugin.

## Testing Done

- [x] Manual testing completed
- [x] Automated tests pass locally (focused suite; full `npm test` needs a local install of repo dependencies)
- [x] Edge cases considered and tested

Focused local tests:

- `node tests/plugin-manifest.test.js` (80 pass)
- `node tests/lib/resolve-ecc-root.test.js` (39 pass)
- `node tests/hooks/hook-flags.test.js` (64 pass)
- `node tests/hooks/plugin-hook-bootstrap.test.js` (15 pass)
- `node tests/hooks/hooks.test.js` (249 pass)
- `node tests/ci/catalog.test.js` (4 pass)
- `node tests/docs/harness-adapter-compliance.test.js` (10 pass)
- `node scripts/ci/catalog.js --text`

Manual Grok CLI (1.0.5):

```text
grok plugin validate .
grok plugin marketplace add /path/to/ECC
grok plugin install /path/to/ECC --trust
grok plugin enable ecc
grok inspect   # ecc enabled: 286 skills, hooks, chrome-devtools MCP
```

`resolveEccRoot()` found `~/.grok/installed-plugins/ecc-*` with no env set. Hook bootstrap with `GROK_PLUGIN_ROOT` set both Grok and Claude plugin-root aliases.

## Type of Change

- [x] `feat:` New feature

## Security & Quality Checklist

- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [ ] Shell scripts pass shellcheck (if applicable)
- [ ] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## If you changed dependencies or `package.json` (`bin` / `files` / deps)

- [x] Ran `yarn install --mode=update-lockfile` and committed the `yarn.lock` change. CI runs Yarn in hardened mode on public PRs and fails if the lockfile would be modified, so an out of date `yarn.lock` breaks the build even when nothing else is wrong.

`package.json` `files` only gained `.grok-plugin/`. No dependency or lockfile change.

## If you added a skill, command, agent, hook, or CLI tool

Not applicable (no new skill/command/agent). Existing hook JSON commands now also read `GROK_PLUGIN_ROOT`.

## Documentation

- [x] Updated relevant documentation
- [x] Added comments for complex logic
- [x] README updated (if needed)

## Notes / limits (intentional)

- Grok plugins stay off until `enable`; hooks/MCP need `--trust`.
- Claude `userConfig` hook profiles are ignored; use `ECC_HOOK_PROFILE`.
- Trusted Grok installs attach root `.mcp.json` (Claude plugin installs opt that out).
- Agent `tools:` lists use Claude names; Grok maps them by tool kind. Nested spawn is still depth-1.
- No `ecc install --target grok` in this PR.

### CODE REVIEW

File: hooks/codex-hooks.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
...
...hrow new Error('Missing Codex PLUGIN_ROOT');process.env.CLAUDE_PLUGIN_ROOT=process.env.PLUGIN_ROOT;const p=require('path');const r=(function(){var p=require('path'),f=require('fs'),o=require('os');var e=process.env.GROK_PLUGIN_ROOT||process.env.CLAUDE_PLUGIN_ROOT||process.env.ECC_PLUGIN_ROOT;if(e&&e.trim())return e.trim();var d=p.join(o.homedir(),'.claude');function L(x){try{return require(p.join(x,'scripts','lib','resolve-ecc-root')).resolveEccRoot()}catch(_){return null}}var r=L(d);if(r)return r;var s=['ecc','ecc@ecc','marketplaces/ecc','everything-claude-code','everything-claude-code@everything-claude-code','marketplaces/everything-claude-code'];for(var i=0;i<s.length;i++){r=L(p.join(d,'plugins',s[i]));if(r)return r}try{var g=['ecc','everything-claude-code'];for(var j=0;j<g.length;j++){var c=p.join(d,'plugins','cache',g[j]);var O=f.readdirSync(c);for(var k=0;k<O.length;k++){var q=p.join(c,O[k]);var V=f.readdirSync(q);for(var m=0;m<V.length;m++){r=L(p.join(q,V[m]));if(r)return r}}}}catch(_){}return d})();const s=p.join(r,'scripts/hooks/plugin-hook-bootstrap.js');process.env.CLAUDE_PLUGIN_ROOT=r;process.argv.splice(1,0,s);require(s)\" node scripts/hooks/session-start-bootstrap.js"
```
[the remark this answers is not in the document]
Comment (Misty-Star) [reply hooks/codex-hooks.json]: Fixed in 8143ec31. Shared root resolution is now `PLUGIN_ROOT` → `CLAUDE_PLUGIN_ROOT` → `GROK_PLUGIN_ROOT` → `ECC_PLUGIN_ROOT`, so a Codex SessionStart that maps `PLUGIN_ROOT` onto `CLAUDE_PLUGIN_ROOT` is not overridden by a leftover Grok install. Regression coverage is in `resolve-ecc-root`, hook-flags, and plugin-hook-bootstrap tests.

**Comment by haelyra** [comment]
Thank you for doing the real Grok CLI validation and for documenting the trust and MCP behavior clearly. A Grok adapter is directionally useful for the provider-neutral ECC roadmap, and I am sorry this waited without a maintainer response.

This version needs a focused architecture pass before merge because provider-specific behavior currently leaks through the core hook, root-resolution, command, localization, and catalog surfaces. It also makes a trusted install attach the repository root `.mcp.json`, including Chrome DevTools, as an incidental consequence of installation. ECC must keep context, capability consent, installation, and evidence separate.

A mergeable adapter should:

- define Grok through the canonical harness capability and adapter contract, keeping `GROK_PLUGIN_ROOT` resolution and Grok layout rules inside that adapter rather than adding aliases throughout shared hook code;
- produce a previewable, receipt-backed install plan with explicit consent for hooks and each MCP capability, with no implicit Chrome DevTools attachment;
- prove source identity, version pinning, root containment, upgrade, uninstall, and rollback for the whole-repository marketplace source;
- add adapter conformance fixtures for untrusted and trusted installs, disabled plugins, missing roots, multiple cached versions, Windows paths, and capability denial;
- keep generated/localized documentation changes out of the first behavioral patch unless the repository generators produce them from the final adapter contract.

That routes the work into the ECC 2 distribution substrate and the ECC 3 replaceable-harness boundary without making Grok another source of policy or canonical state. If you do not have bandwidth for the adapter refactor, please let us know and we will carry the validated Grok findings into the post-release harness-adapter queue with attribution. We appreciate the careful exploratory work here.

**Comment by Misty-Star** [comment]
Thanks for the architecture direction, and for taking the time to write this up after the delay.

I have bandwidth for the adapter refactor and will rework this PR against that contract instead of merging the current leak-through version.

My reading of a mergeable Grok adapter:

- Keep Grok-specific layout and `GROK_PLUGIN_ROOT` resolution inside the harness adapter. Shared hook, root-resolution, command, catalog, and localization surfaces should not grow Grok aliases.
- Make install previewable and receipt-backed, with explicit consent for hooks and each MCP capability. Trusted install must not implicitly attach root `.mcp.json` / Chrome DevTools.
- Prove source identity, version pinning, root containment, upgrade, uninstall, and rollback for the whole-repository marketplace source.
- Add adapter conformance fixtures for untrusted/trusted installs, disabled plugins, missing roots, multiple cached versions, Windows paths, and capability denial.
- Keep generated/localized docs out of the first behavioral patch unless they are produced from the final adapter contract.

I’ll treat Grok as a replaceable harness adapter on the ECC 2 distribution substrate / ECC 3 harness boundary, not as a new source of policy or canonical state.

I’ll start with the adapter contract and conformance fixtures, then fold the already-validated Grok CLI findings (`--trust`, MCP attach, root discovery) into that shape. I’ll comment again when the first behavioral patch is ready for review.

File: .grok-plugin/plugin.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "skills",
+    "hooks",
+    "tdd",
+    "code-review",
+    "security",
+    "workflow"
+  ],
+  "logo": "assets/ecc-icon.svg",
+  "mcpServers": "",
+  "hooks": ""
```
[the remark this answers is not in the document]
Comment (Misty-Star) [reply .grok-plugin/plugin.json:24]: Thanks for flagging this. I verified the behavior against Grok Build commit 72a61251fcffb464bcc687aeb5a998e5a98ec0c9.

For `PathOrInline`, an explicitly present empty string is resolved as a path relative to the plugin root. That resolves to the plugin root directory, which is not a file, so the component resolves to `None`. Convention-based fallback only occurs when the manifest field is absent.

Relevant implementation:
  https://github.com/xai-org/grok-build/blob/72a61251fcffb464bcc687aeb5a998e5a98ec0c9/crates/codegen/xai-grok-agent/src/plugins/manifest.rs#L80-L110
  https://github.com/xai-org/grok-build/blob/72a61251fcffb464bcc687aeb5a998e5a98ec0c9/crates/codegen/xai-grok-agent/src/plugins/manifest.rs#L182-L192
  https://github.com/xai-org/grok-build/blob/72a61251fcffb464bcc687aeb5a998e5a98ec0c9/crates/codegen/xai-grok-agent/src/plugins/discovery.rs#L679-L702

I also ran `grok plugin validate .`; it reported only the skill, command, and agent components, with no hooks or MCP servers. Therefore `mcpServers: ""` and `hooks: ""` are the currently supported explicit opt-out, while omitting the fields would re-enable convention-based discovery. Keeping the existing values.

File: scripts/lib/install-targets/grok-home.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            ...(normalizeRelativePath(sourceRelativePath) === 'hooks/hooks.json'
+              ? { contentTransform: 'grok-hook-boundary' }
+              : {}),
+          };
+        });
+    });
+    const hooksEnabled = allowHooks && operations.some((operation) => isHooksPath(operation.sourceRelativePath));
+    const pluginRoot = path.join(adapter.resolveRoot(input), 'plugins', 'ecc');
+    const hooksManifestPath = path.join(pluginRoot, 'hooks', 'hooks.json');
+    if (hooksEnabled && !operations.some((operation) => operation.destinationPath === hooksManifestPath)) {
```
[the remark this answers is not in the document]
Comment (Misty-Star) [reply scripts/lib/install-targets/grok-home.js:94]: Verified against the canonical install pipeline. `grok-home.planOperations()` can emit the directory scaffold plus the explicit manifest scaffold, but these are not written directly or recorded as install-state ownership. `createManifestInstallPlan()` first materializes directory scaffolds into per-file `copy-file` operations (`scripts/lib/install/plan.js:277-279`), then `dedupeCopyFileOperations()` keeps only the last copy for each destination (`scripts/lib/install/plan.js:208-231`, invoked at lines 281-287). The explicit `hooks/hooks.json` operation is appended after the directory operation and carries `contentTransform: "grok-hook-boundary"`, so it is the retained operation. I also reproduced the `paths: ["hooks"]` case through `createManifestInstallPlan`: the final plan and `statePreview` contain exactly one destination ending in `hooks/hooks.json`, with `grok-hook-boundary`. Therefore there is no overlapping final ownership or untransformed manifest write to fix here; changing the raw scaffold representation would add complexity without changing install behavior.

File: .grok-plugin/marketplace.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        "skills",
+        "hooks",
+        "tdd",
+        "code-review"
+      ],
+      "category": "development",
+      "source": {
+        "source": "url",
+        "url": "https://github.com/affaan-m/ECC.git",
+        "sha": "b09ccb63706ab60787b9993bbb4c14df1291867b"
```
[the remark this answers is not in the document]
Comment (Misty-Star) [reply .grok-plugin/marketplace.json]: Thanks for the concrete ancestry check. This is an intentional pre-merge self-pin, not a floating ref:

- The manifest uses the full immutable SHA `b09ccb63706ab60787b9993bbb4c14df1291867b`, which is reachable from this PR head `c8d4f01c60b2c58ae04bf77b0599773b2d77c23b` and fetches successfully.
- The current `main` (`e04ea0b9cc8248686edf5ac751cadff550e162b8`) does not contain `.grok-plugin/marketplace.json` or `scripts/lib/grok-harness-adapter.js`. Pinning `main` or `v2.2.0` would make the marketplace install code without the Grok feature.
- Before this PR is merged, a commit containing these changes cannot also be an ancestor of `main`; that is an expected property of a self-referencing marketplace entry. The SHA remains directly fetchable and auditable by content.

After merge, the pin should be updated to the canonical merge/release commit if the repository merge strategy rewrites the PR commits (for example, squash). There is no safe in-PR repin that satisfies main ancestry without removing the feature. Please classify this as an expected pre-merge lifecycle condition or defer the canonical repin until merge/release.

**Comment by Misty-Star** [comment]
@haelyra The focused adapter pass is now complete on `c8d4f01`.

Grok-specific layout and root discovery stay within the harness adapter, while the shared install planner carries previewable, receipt-backed operations with explicit hook and per-capability MCP consent. The canonical lifecycle paths cover pinned source identity, root containment, upgrade, repair, rollback, and uninstall; the conformance coverage includes trust/denial, disabled plugins, missing roots, multiple cached versions, and Windows paths.

Validation is green for the focused surfaces: install/apply (43), canonical Grok install (10), plugin manifest (80), install targets (53), and install executor (16), with the broader suite and adapter harness checks also passing. Greptile also withdrew its marketplace ancestry finding as an expected pre-merge lifecycle condition, and the review thread is resolved.

File: scripts/lib/grok-harness-adapter.js [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  return null;
+}
+
+function resolveGrokPluginRoot(options = {}) {
+  const pathModule = options.pathModule || path;
+  if (options.enabled === false) return null;
+  const envRoot = options.envRoot !== undefined
+    ? trimEnv(options.envRoot)
+    : grokPluginRootFromEnv(options.env || {});
+  if (envRoot) return envRoot;
```
[the remark this answers is not in the document]
Comment (Misty-Star) [reply scripts/lib/grok-harness-adapter.js]: Fixed in `459c371e56ade2a3ee0b4d4ae828f1fc57e34393` and pinned by `93856648bbfabfcb1bbd1137239913f26bc89564`.

When a pin is present, an explicit `GROK_PLUGIN_ROOT` is now accepted only if the root is complete and its parseable `.ecc-source.json` has an exact matching SHA; otherwise resolution fails closed with `null`. Explicit roots without a pin retain the existing compatibility behavior.

Regression coverage now proves both mismatched-root rejection and matching-root acceptance. Validation passed: Grok adapter 8/8, canonical install 10/10, plugin manifest 80/80, install lifecycle 66/66, and install executor 20/20.

**Comment by chenhz01** [comment]
The adapter refactor on `c8d4f01` addresses the leak-through @haelyra flagged. One manifest-level issue survives the rework: `.grok-plugin/marketplace.json` pins `source.sha` to a fixed commit, so marketplace installs resolve to that frozen tree — the pin goes stale on the first merge to main after this lands, and updating it is a manual edit nobody owns. Either drop the `sha` and let Grok resolve the URL head, or own it in CI:
```yaml
- run: node scripts/ci/update-grok-pin.js && git diff --exit-code
```
so a stale pin fails the build instead of silently freezing users on an old ECC. Happy to open a PR with working code if useful.
